pkg/k8s/apis/cilium.io/v2alpha1/bgp_node_types.go

```diff
@@ -0,0 +1,183 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package v2alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:resource:categories={cilium,ciliumbgp},singular="ciliumbgpnodeconfig",path="ciliumbgpnodeconfigs",scope="Cluster",shortName={bgpnode}
```

Comment:
@youngnick Do you have any opinion about the `shortName` naming convention? You mentioned that the `Cilium` prefix in the full name helps avoid naming collision, but we don't have a cilium prefix here for short names.

Reply:
Yes, I think that maybe `cbgpnode` might be slightly better, although it will suck to type.

Reply:
@harsimran-pabla I think we should rename or add one more shortName. Especially, `bgppeer` is colliding with the MetalLB's `BGPPeer` CRD's name, so if users install MetalLB and Cilium at the same time, it may cause unexpected behavior.

Reply:
@YutaroHayakawa I updated the short names to prefix them with 'c'.